### PR TITLE
refine HttpServer.h

### PR DESCRIPTION
**Pull request by ian287913**

### CODE REVIEW

File: src/HttpServer.h [hunk 1]
```diff
@@ -5,48 +5,71 @@ All rights reserved.
...
-  // Start the http server
+  // Start the server
   void Start();
-  // Close the http server
+  // Close all the threads and async-works
   void Shutdown();
   // Get the server status
-  int GetServerStatus();
-  // Get the current number of clients
+  ServerStatus GetServerStatus() { return serverStatus; }
```
Comment (firewings2515) [inline src/HttpServer.h:48]: use const member function to getfunc.

File: src/HttpServer.h [hunk 2]
```diff
@@ -5,48 +5,71 @@ All rights reserved.
...
+  // Get the number of current serving clients
   int GetClientCount();
-  // Determines the size of the listenerThreadPool
-  const int maxClientCount = 10;
+  // Get the size of the thread pool
+  int GetThreadPoolSize() { return SocketHandlerPool->GetMaxPoolSize(); }
```
Comment (firewings2515) [inline src/HttpServer.h:52]: use const member function to getfunc.

File: src/HttpServer.h [hunk 3]
```diff
@@ -5,48 +5,71 @@ All rights reserved.
...
   void Start();
-  // Close the http server
+  // Close all the threads and async-works
   void Shutdown();
   // Get the server status
-  int GetServerStatus();
-  // Get the current number of clients
+  ServerStatus GetServerStatus() { return serverStatus; }
+  // Get the number of current serving clients
   int GetClientCount();
```
Comment (firewings2515) [inline src/HttpServer.h:50]: if the implement will not change the member in class, use const member function to get function

File: src/HttpServer.h [hunk 4]
```diff
@@ -5,48 +5,71 @@ All rights reserved.
...
 found in the LICENSE file. See the AUTHORS file for names of contributors.
  */
+#include <boost/array.hpp>
+#include <boost/asio.hpp>
+#include <boost/bind/bind.hpp>
+#include <boost/enable_shared_from_this.hpp>
+#include <boost/shared_ptr.hpp>
+#include <ctime>
 #include <iostream>
+#include <string>
```
[the remark this answers is not in the document]
Comment (john0312) [reply src/HttpServer.h:15]: This is a reasonable review.

File: src/HttpServer.h (hunk 3 above)
Comment (john0312) [reply src/HttpServer.h:50]: This is a reasonable review.

File: src/HttpServer.h (hunk 2 above)
Comment (john0312) [reply src/HttpServer.h:52]: This is a reasonable review.

File: src/HttpServer.h [hunk 5]
```diff
@@ -5,48 +5,71 @@ All rights reserved.
...
+  // Thread pool initialization. This will be called by the constructor.
+  void InitializeSocketHandlerPool(int size);
+  // Start to listen to incoming client connections
+  void StartAccept();
+  // Handle the connected socket. Hand over the socket to a working thread and
+  // call StartAccept() to keep waiting for another connection. (This function
+  // will be autometically called after async_accept() finishes)
+  void HandleAccept(const boost::system::error_code& error);
+  //	io_context for async works
+  boost::asio::io_context& ioContext;
```
Comment (john0312) [inline src/HttpServer.h:64]: Please fix naming, same below.

File: src/HttpServer.h [hunk 6]
```diff
@@ -5,48 +5,71 @@ All rights reserved.
...
+#include <string>
 
 #include "ThreadPool.h"
 
 #ifndef SHORTLINK_SRC_HTTPSERVER_H_
 #define SHORTLINK_SRC_HTTPSERVER_H_
 
 namespace shortlink {
-
+enum ServerStatus { kRunning, kStopped };
```
Comment (john0312) [inline src/HttpServer.h:23]: Please add a comment on what this is for.

File: src/HttpServer.h (hunk 5 above)
Comment (YuanWeiChen16) [reply src/HttpServer.h:64]: please reference
https://google.github.io/styleguide/cppguide.html#Variable_Names

**Review by YuanWeiChen16 — COMMENTED** [review]
viewed

File: src/HttpServer.h (hunk 4 above)
[the remark this answers is not in the document]
Comment (ian287913) [reply src/HttpServer.h]: Include re-ordered.
It seems like clang-format will re-order the include lines from:
`
#include <ctime>
#include <iostream>
#include <string>

#include <boost/array.hpp>
#include <boost/asio.hpp>
#include <boost/bind/bind.hpp>
#include <boost/enable_shared_from_this.hpp>
#include <boost/shared_ptr.hpp>

#include "ThreadPool.h"`
to:
`
#include <boost/array.hpp>
#include <boost/asio.hpp>
#include <boost/bind/bind.hpp>
#include <boost/enable_shared_from_this.hpp>
#include <boost/shared_ptr.hpp>
#include <ctime>
#include <iostream>
#include <string>

#include "ThreadPool.h"`
is this normal?

File: src/HttpServer.h (hunk 2 above)
Comment (ian287913) [reply src/HttpServer.h]: const keyword added.

File: src/HttpServer.h (hunk 3 above)
Comment (ian287913) [reply src/HttpServer.h]: const keyword added.

File: src/HttpServer.h (hunk 5 above)
Comment (ian287913) [reply src/HttpServer.h]: naming fixed.

File: src/HttpServer.h (hunk 6 above)
Comment (ian287913) [reply src/HttpServer.h:27]: comment added.

File: src/HttpServer.h (hunk 5 above)
Comment (ian287913) [reply src/HttpServer.h]: naming fixed.

**Review by john0312 — APPROVED** [review]
Please ensure your header ordering is consistent across the project before merging.

**Review by younger15 — APPROVED** [review]

**Pull request merged**